core/src/main/scala/org/bitcoins/core/wallet/utxo/InputInfo.scala

```diff
@@ -110,6 +112,53 @@ object InputInfo {
     }
   }
 
+  /**
+    * Returns the needed hash pre-images and conditional path that was used to spend the input
+    * at inputIndex, this is calculated through the ScriptSignature and ScriptWitness
+    */
+  def getHashPreImagesAndConditionalPath(
+      signedTransaction: Transaction,
+      inputIndex: Int): (Vector[NetworkElement], ConditionalPath) = {
+
+    val txIn = signedTransaction.inputs(inputIndex)
+
+    @tailrec
+    def getPreImagesAndCondPath(
+        scriptSignature: ScriptSignature,
+        prevCond: ConditionalPath = ConditionalPath.NoCondition): (
```

Comment:
More efficient to take a `Vector[Boolean]` (default to `empty`) and only convert back outside (rather than doing in and out of the type on every iteration)